structure_tests/README.md

```diff
@@ -0,0 +1,53 @@
+Container Build Structure Tests
+===============================
+
+This code builds an image which serves as a framework to run structure-based tests on an image prior to pushing it to GCR. The image under tests runs as a docker container **inside** of this image, which itself runs as a docker container on a host machine in the cloud (when run through a [Google Cloud Container Build](https://cloud.google.com/container-builder/docs/overview)).
+
+To use this test image with any cloudbuild, add the following build step to the **end** your container build config (cloudbuild.yaml or cloudbuild.json):
+
+	name: gcr.io/gcp-runtimes/structure_test
+	args:
+		- <your_target_image>
+
+It's **very important that this step appears at the end of your build** (or at least after the image itself it assembled by Docker); without a built image, there will be nothing to test, and your build will fail!
+
+Tests within this framework are specified through a JSON config file, by default called `structure_test.json` (this can be specified through a `--config` flag argument to the build step). This file will be copied into the workspace of the structure test image and loaded in by the test driver, which will execute the tests in order. Within this config file, three distinct types of tests can be written:
+
+- Command Tests (testing output/error of a specific command issued)
+- File Existence Tests (making sure a file is, or isn't, present in the file system of the image)
+- File Content Tests (making sure files in the file system of the image contain, or do not contain, specific contents)
+
+## Command Tests
+Command tests ensure that certain commands run properly on top of the shell of the target image. Regexes can be used to check for expected or excluded strings in both stdout and stderr. Additionally, any number of flags can be passed to the argument as normal.
+
+####Supported JSON Fields:
+
+- Name (string, **required**): The name of the test
```

Comment:
Something like this project might be nice: https://www.npmjs.com/package/json-schema-to-markdown

We could generate a JSON schema, and then turn that into Markdown.